nilearn/tests/test_datasets.py

```diff
@@ -27,14 +27,22 @@
 file_mock = None
 
 
+def setup_silence():
+    # Wrap functions with versions that have verbose=0
+    datasets._get_dataset_dir = silence_fn(datasets._get_dataset_dir)
+    datasets._uncompress_file = silence_fn(datasets._uncompress_file)
+    datasets._fetch_file = silence_fn(datasets._fetch_file)
+    datasets._fetch_files = silence_fn(datasets._fetch_files)
+    datasets._chunk_read_ = silence_fn(datasets._chunk_read_)
```

Comment:
I am sligthly uneasy with this for two reasons:
* you are silencing all functions at once, which means that you could hide potential issues, for example if a verbose parameter wasn't passed from the caller to the callee (you fixed a few of those in this PR by the way, thanks for that)
* ideally, it'd be good to have a teardown equivalent of this that sets back the datasets functions to their original selves, so that each test doesn't modify the global state. I am guessing that with the current version of the code, you only need one decorated test to silence all of them.